### DISCUSSION TITLE
LXTerminal command prompts lack color

### DISCUSSION THREAD

**Discussion by dshaw619** [gitbls/sdm#56] [Q&A] [answered]
Not sure what might be going on here or if I am doing something wrong.  I have found that images customized and burned with sdm don't display the command prompts in LXTerminal in color as they normally do.  Everything is just monochrome including the different file types displayed by 'ls'.  

I was using a lxterminal.conf file to set font size and scrollback lines, but the problem persists even if I remove that switch from sdm-customize.  

I rechecked that I hadn't somehow used a Buster conf file (there are a couple of differences in the Preferences options) and recopied the lxterminal.conf file from a running Bullseye system.  Restarted LXTerminal and see same behaviour. 

I flashed a brand new Bullseye image from raspberry.com using RPi Imager and it show the colors as expected.  

When using the sdm images there is also a space missing before the '$' in the command prompt that appears on all my other systems.  I can't demonstrate the color differences here, but this difference in the prompt is shown below.  Strawberry is another Bullseye system previously flashed via Imager.  KeyLime was customized and burned via sdm.
`des@strawberry:~ $ `
`des@KeyLime:~$ `

Is sdm somehow affecting the LXTerminal settings or did I miss something?

**Comment by gitbls** [comment]
sdm doesn't do ANYTHING relative to lxterminal unless you explicitly tell it to using `--lxde-config`.

Can you provide the sdm command line and any lxde config-related files you're using so that I can have a look at it? Without a repro, I can only guess that you have left a file in /home/{username}/.config/lxterminal that is causing this.

**Comment by gitbls** [comment]
Took a look and found that the color prompt is set by ~username/.bashrc. When the Pi software creates a new user (for instance, when the Pi first boots), it copies a few files (.bashrc, .bash_logout, and .profile) into the user directory from /etc/skel/.bashrc.

sdm doesn't do that; will fix.

**Reply by dshaw619** [reply]
I just customized and burned another image and now the color prompts, etc. are working!  I haven't installed any changes you might have made since v7.6, so puzzled.  Only things I can think of that I did different are that in the customize command, I replaced '--user des' with '--nouser' and in the burn command I added '--redact'.  Both of these commands are run from within scripts that I use to ensure consistency.

```
sudo /usr/local/sdm/sdm --customize \
--wpa $HOME/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/wpa_supplicant.conf \
--keymap us \
--locale en_US.UTF-8 \
--timezone America/Los_Angeles \
--wifi-country US \
--disable piwiz \
--svc-disable userconfig \
--bootset serial=0,boot_behaviour=B4,boot_wait=0 \
--nouser \
--fstab $HOME/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/fstab_append.txt \
--extend --xmb 1024 \
--poptions noupdate,noupgrade \
--restart \
~/Downloads/bullseye-armhf-customized.img
```
```
sudo /usr/local/sdm/sdm --burn /dev/sdd \
--ddsw bs=16M \
--hostname KeyLime \
--user des \
--redact \
--plugin vnc:"realvnc=1280x720" \
--expand-root \
~/Downloads/bullseye-armhf-customized.img
```

**Comment by gitbls** [comment, marked as answer]
Yes, this is correct. During a customize, adding a user is done with `useradd --no-create-home` because I (or maybe you) may want to copy files into your login directory from the running system during a custom phase script or plugin Phase 0.

If the home directory exists `useradd --create-home` will not copy the /etc/skel files into it. SO, in the Phase0/Phase1 case (e.g., `--user bls`), the /etc/skel files don't get copied, and the result is that your prompts aren't colored. I've updated sdm (not checked in yet) to explicitly copy the /etc/skel files (.profile, .bashrc, .bash_logout) that don't exist in the login directory.

However, in the case of `--burn --user des`, sdm IS able to use `useradd --create-home`, which puts the /etc/skel files into /home/des. 

It's definitely an oddity, but once this update is checked in, the behavior will be consistent between the two. Except in the case of someone who wants to use /etc/skel to add additional files to the login directory. That won't get picked up in Phase0/Phase1 because I don't currently process all the files in /etc/skel. I could, but I think I'll wait on that for now.

**Reply by dshaw619** [reply]
Thanks for that detailed explanation.

**Reply by dshaw619** [reply]
Forgot to add:  Looking at /etc/skel on a freshly configured Bullseye system I seen another file (.mkshrc) that you don't mention.  It seems to be related to YAS (Yet Another Shell (:-) ) which I don't expect to ever need myself, but for completeness sake you may want to include.

**Reply by gitbls** [reply]
yea, I'm going to change the code to just copy all the files that are in /etc/skel, although interestingly using useradd doesn't copy the .mkshrc file.
